src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -1135,13 +1081,15 @@ void Leader::RemoveRloc(uint16_t aRloc16, MatchMode aMatchMode)
         {
         case NetworkDataTlv::kTypePrefix:
         {
-            PrefixTlv *prefix = static_cast<PrefixTlv *>(cur);
+            PrefixTlv *      prefix = static_cast<PrefixTlv *>(cur);
+            const PrefixTlv *excludePrefix =
+                FindPrefix(prefix->GetPrefix(), prefix->GetPrefixLength(), aExcludeTlvs, aExcludeTlvsLength);
 
-            RemoveRloc(*prefix, aRloc16, aMatchMode);
+            RemoveRlocInPrefix(*prefix, aRloc16, aMatchMode, excludePrefix, aChangedFlags);
 
-            if (prefix->GetSubTlvsLength() == 0)
+            if (UpdatePrefix(*prefix) == kTlvRemoved)
             {
-                RemoveTlv(prefix);
+                // Do not update `cur` when TLV is removed.
```

Comment:
Not introduced by this PR, but this skips the log message below. Should we improve the logging?

Reply:
I agree.  I actually prefer if we remove some of the debg level logs (e.g. from `RemoveRloc()` we just log at the end after all related TLVs/sub-TLvs/entries are removed (and not log per prefix/entry). We already do the same when adding new entries. What do you think?

Reply:
Sounds good to me!

Reply:
Changed this in the latest pushed commit.